appsec/src/helper/subscriber/waf.cpp

```diff
@@ -336,8 +382,76 @@ void load_result_report(
     }
 }
 
+void handle_config_diagnostics(const remote_config::parsed_config_key &cfg_key,
+    parameter_view diagnostics, telemetry::telemetry_submitter &msubmitter)
+{
+    using log_level = telemetry::telemetry_submitter::log_level;
+
+    const auto info = static_cast<parameter_view::map>(diagnostics);
+    for (auto k : diagnostic_keys) {
+        auto it = info.find(k);
+        if (it == info.end()) {
+            continue;
+        }
+
+        const parameter_view &value = it->second;
+        if (!value.is_map()) {
+            continue;
+        }
+
+        // map has "error", "loaded", "failed", "skipped", "errors", "warnings"
+        const parameter_view::map map = static_cast<parameter_view::map>(value);
+
+        if (map.contains("error")) {
+            auto message = static_cast<std::string>(map.at("error"));
+            std::string identifier = fmt::format(
+                "rc::{}::exception", cfg_key.product().name_lower());
```

Comment:
Diagnostics are sent as `rc::{}::diagnostic` on error or warning level (instead of `::exception` for error and `::diagnostic` for warning).